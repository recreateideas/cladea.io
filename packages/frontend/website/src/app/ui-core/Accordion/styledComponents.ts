import styled, { css } from 'styled-components';
import {
    Accordion as MUIAccordion,
    AccordionDetails as MUIAccordionDetails,
    AccordionSummary as MUIAccordionSummary,
} from '@mui/material';

export const Container = styled.div``;
export const Accordion = styled(MUIAccordion)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            background-color: ${palette.neutrals.secondaryBg};
            box-shadow: none;
            background-image: none;
        `;
    }}
`;
export const AccordionDetails = styled(MUIAccordionDetails)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            padding: 0px ${padding.small} ${padding.small} ${padding.small};
        `;
    }}
`;
export const AccordionSummary = styled(MUIAccordionSummary)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    sizes,
                    layout: { padding },
                },
            },
        } = props;
        return css`
            &.Mui-expanded {
                min-height: ${sizes.accordion.medium};
            }
            > div {
                margin: ${padding.small} 0 !important;
            }
            padding: 0px ${padding.small};
        `;
    }}
`;
export const Title = styled.div``;
