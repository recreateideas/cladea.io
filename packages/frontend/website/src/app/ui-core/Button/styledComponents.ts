import styled, { css } from 'styled-components';
import { Button } from '@mui/material';

export const StyledButton = styled(Button)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            &.MuiButton-containedPrimary .MuiButton-label {
                color: ${palette.fonts.primaryInverted};
            }
        `;
    }}
`;
