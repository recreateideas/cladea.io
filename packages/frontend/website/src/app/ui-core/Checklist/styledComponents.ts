import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    ul {
        background-color: inherit;
        .MuiListItemIcon-root {
            min-width: auto;
        }
        .MuiButtonBase-root {
        }
    }
`;
