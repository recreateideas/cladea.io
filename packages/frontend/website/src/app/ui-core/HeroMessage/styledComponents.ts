import styled, { css } from 'styled-components';

export const Container = styled.div<{ type: 'error' | 'warn' | 'success' }>`
    ${(props) => {
        const {
            theme: {
                dsl: { hexToRgba, styles, layout, palette },
            },
            type,
        } = props;
        const colorMap: {
            [type: string]: {
                color: string;
                backgroundColor: string;
            };
        } = {
            error: {
                color: palette.fonts.error,
                backgroundColor: hexToRgba(palette.secondary.orange[500], 10),
            },
            success: {
                color: palette.fonts.success,
                backgroundColor: hexToRgba(palette.secondary.green[500], 10),
            },
            warn: {
                color: palette.fonts.warn,
                backgroundColor: hexToRgba(palette.secondary.yellow[500], 10),
            },
        };
        return css`
            ${colorMap[type]}
            white-space: pre;
            padding: ${layout.padding.small};
            border-radius: ${styles.borderRadius};
            overflow-x: auto;
            .section {
                display: flex;
                .icon {
                    margin-right: ${layout.margin.small};
                }
            }
        `;
    }}
`;
