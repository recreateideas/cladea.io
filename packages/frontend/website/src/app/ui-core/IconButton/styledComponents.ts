import styled, { css } from 'styled-components';

export const Container = styled.div<{ shape?: 'square'; disabled?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            disabled,
            shape,
        } = props;
        const bordersRadius = shape === 'square' ? styles.borderRadius : undefined;
        return css`
            ${bordersRadius &&
            css`
                .MuiButtonBase-root {
                    border-radius: ${bordersRadius};
                }
            `}
            ${disabled &&
            css`
                &,
                * {
                    cursor: no-drop;
                    svg {
                        fill: ${palette.neutrals.inactive} !important;
                    }
                }
            `}
        `;
    }}
`;
