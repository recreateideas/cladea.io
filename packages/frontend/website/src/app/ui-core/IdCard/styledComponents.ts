import styled, { css } from "styled-components";
import { avatar } from "../images";

export const Container = styled.div<{ breakpoint: string }>`
  ${(props) => {
    const {
      theme: {
        dsl: { layout, palette },
      },
      breakpoint,
    } = props;
    const isSmall = breakpoint === "sm";
    const width = isSmall ? "100%" : "30vw";
    return css`
      border-radius: 12px;
      min-width: ${width};
      background-color: ${palette.neutrals.quaternaryBg};
      border: solid 1px ${palette.neutrals.borders};
      overflow: hidden;
      .background {
        width: 100%;
        height: 60vh;
        fill: ${palette.neutrals.secondaryBg};
      }
      z-index: ${layout.namedZIndex.leftSidebar - 1};
      .bio {
        display: flex;
        flex-direction: column;
      }
      .avatar {
        position: relative;
        border-radius: 100%;
        overflow: hidden;
        background: ${palette.neutrals.primaryBg};
        width: 20vh;
        height: 20vh;
        margin: 24px auto;
        .image-container {
          border-radius: 100%;
          overflow: hidden;
          width: inherit;
          height: inherit;
          background-image: url(${avatar});
          background-position: center;
          background-repeat: no-repeat;
          background-size: cover;
        }
        .space {
          background: unset !important;
        }
      }
      .name {
        display: flex;
        letter-spacing: 1px;
        justify-content: center;
        gap: 16px;
        .first {
        }
        .last {
          color: ${palette.tertiary.pink[500]};
        }
      }
      .position {
        width: 100%;
        .title {
          color: ${palette.secondary.neon[500]};
          text-align: center;
        }
      }
      .contacts {
        display: flex;
        flex-direction: column;
        gap: 16px;
        margin: ${isSmall ? "2vh" : "24px"} 24px;
        padding: ${isSmall ? "2vh" : "24px"};
        border: solid 1px ${palette.neutrals.borders};
        border-radius: 8px;
        background-color: ${palette.neutrals.secondaryBg};
        .contact-container {
          width: 100%;
          /* height: 40px; */
          display: flex;
          justify-content: center;
          .href-container {
            text-decoration: none;
            display: flex;
            gap: 16px;
            color: ${palette.fonts.primary};
            cursor: pointer;
            img {
              width: 36px;
            }
            .href-text {
              display: flex;
              flex-direction: column;
              justify-content: center;
              width: 80px;
            }
          }
        }
      }
    `;
  }};
`;

export const ToastContainer = styled.div`
  ${(props) => {
    const {
      theme: {
        dsl: { palette },
      },
    } = props;
    return css`
      .toast {
        display: block;
        position: fixed;
        top: 50%;
        left: 50%;
        transform: translateX(-50%);
        min-width: 450px;
        background-color: ${palette.neutrals.quaternaryBg};
        box-shadow: ${palette.neutrals.boxShadow(palette.secondary.neon[500])};
        border: solid 1px ${palette.neutrals.borders};
        .btn-close {
          color: ${palette.fonts.primary};
          background-color: ${palette.secondary.neon[500]};
        }
        .toast-header {
          justify-content: space-between;
          background-color: ${palette.neutrals.primaryBg};
          button {
            margin-right: 0;
          }
          .header-title {
            font-weight: 600;
            width: 100%;
            color: ${palette.secondary.neon[500]};
          }
        }

        .body-content {
          width: 100%;
          display: flex;
          justify-content: space-between;
          .copied-value {
            color: ${palette.tertiary.pink[500]};
          }
        }
      }
    `;
  }};
`;
