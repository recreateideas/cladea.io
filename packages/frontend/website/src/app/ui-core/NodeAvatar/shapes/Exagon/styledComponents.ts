import styled, { css } from 'styled-components';

interface IExagon {
    inner?: boolean;
    backgroundColor: string;
    width: number;
    height: number;
    border: number;
}

const exagon = ({ inner, width, height, border, backgroundColor }: IExagon) => {
    return css`
        ${inner &&
        css`
            left: 1px;
            z-index: 1;
        `}
        position: relative;
        height: ${height}px;
        width: ${width}px;
        background-color: ${backgroundColor};
        margin: ${height / 2}px 0;
        &:before,
        &:after {
            content: '';
            position: absolute;
            width: 0;
            border-left: ${width / 2}px solid transparent;
            border-right: ${width / 2}px solid transparent;
        }
        &:before {
            bottom: 100%;
            border-bottom: ${border}px solid ${backgroundColor};
        }
        &:after {
            top: 100%;
            width: 0;
            border-top: ${border}px solid ${backgroundColor};
        }
    `;
};

export const Container = styled.div<{ size?: 'small' | 'medium'; inner?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    palette,
                    // layout,
                },
            },
            inner,
            size = 'medium',
        } = props;
        const sizes = {
            small: 62,
            medium: 102,
        };
        const length = sizes[size];
        const whRatio = 1.732;
        const width = inner ? length - 2 : length;
        const height = inner ? width / whRatio + 1 : width / whRatio;
        const border = inner ? height / 2 - 1 : height / 2;
        const backgroundColor = inner ? palette.neutrals.secondaryBg : palette.neutrals.borders;
        return css`
            ${exagon({ inner, width, height, border, backgroundColor })}
            .content {
                position: relative;
                top: -1px;
                .icon {
                    display: flex;
                    justify-content: center;
                }
                .header {
                    flex-direction: column;
                    text-align: center;
                }
                .react-flow__handle-bottom {
                    top: ${height +
                    border -
                    2 +
                    1}px; // 1px because content is pushed up 1px in line 69
                    z-index: 2;
                }
            }
        `;
    }}
`;
