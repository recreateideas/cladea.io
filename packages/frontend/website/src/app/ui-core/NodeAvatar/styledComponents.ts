import styled, { css, ThemedStyledProps } from 'styled-components';

interface IContainer {
    size?: 'small' | 'medium';
    isActive?: boolean;
    customCss?: ThemedStyledProps<any, any>;
}

export const DefaultContainer = styled.div<IContainer>`
    box-sizing: border-box;
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            isActive,
            size = 'small',
            customCss,
        } = props;
        const sizeMap: {
            [key: string]: {
                height: number | string;
                width: number | string;
            };
        } = {
            small: {
                width: '100%',
                height: 32,
            },
            medium: {
                width: 256,
                height: 56,
            },
        };
        return css`
            ${isActive &&
            css`
                background-color: ${palette.neutrals.tertiaryBg} !important;
            `}
            background-color: ${palette.neutrals.secondaryBg};
            border: solid 1px ${palette.neutrals.borders};
            border-radius: ${styles.borderRadius};
            ${sizeMap[size]}
            ${customCss}
        `;
    }}
`;

export const Header = styled.div`
    display: flex;
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            .left {
                display: flex;
                &.as-column {
                    flex-direction: column;
                }
            }
            padding: ${padding.xsmall};
            display: flex;
            justify-content: space-between;
        `;
    }}
`;

export const HeaderText = styled.div<{ size?: 'small' | 'medium' }>`
    display: flex;
    flex-direction: column;
    ${(props) => {
        const {
            theme: {
                dsl: {
                    typography: { typefaces },
                    layout: { padding },
                },
            },
            size = 'small',
        } = props;
        const sizes = {
            small: 'caption',
            medium: 'p',
        };
        const nameSize = sizes[size];
        return css`
            padding-left: ${padding.xsmall};
            .name {
                ${typefaces[nameSize]({ color: 'primary' })}
            }
            .type {
                ${typefaces.caption()}
            }
        `;
    }}
`;

export const Body = styled.div``;
