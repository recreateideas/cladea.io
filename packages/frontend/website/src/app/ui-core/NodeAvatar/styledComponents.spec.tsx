import { HeaderText, Header, DefaultContainer } from './styledComponents';
import { TestProvider } from 'src/tests';

const { renderWithProviderAndTheme } = TestProvider({});

describe('styledComponents', () => {
    describe('<HeaderText/>', () => {
        it('should render', () => {
            const root = renderWithProviderAndTheme(<HeaderText />);
            expect(root).toMatchSnapshot();
        });
        it("should render with size='medium'", () => {
            const root = renderWithProviderAndTheme(<HeaderText size="medium" />);
            expect(root).toMatchSnapshot();
        });
    });
    describe('<Header/>', () => {
        it('should render', () => {
            const root = renderWithProviderAndTheme(<Header />);
            expect(root).toMatchSnapshot();
        });
    });
    describe('<DefaultContainer/>', () => {
        it('should render', () => {
            const root = renderWithProviderAndTheme(<DefaultContainer />);
            expect(root).toMatchSnapshot();
        });
        it('should render the active style', () => {
            const root = renderWithProviderAndTheme(<DefaultContainer isActive={true} />);
            expect(root).toMatchSnapshot();
        });
    });
});
