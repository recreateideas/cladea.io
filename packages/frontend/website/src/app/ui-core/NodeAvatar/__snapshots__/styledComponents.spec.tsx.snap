// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styledComponents <DefaultContainer/> should render 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #f9fafb;
  border: solid 1px #dedeeb;
  border-radius: 4px;
  width: 100%;
  height: 32px;
}

<div
  class="c0"
/>
`;

exports[`styledComponents <DefaultContainer/> should render the active style 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #f5f5f8 !important;
  background-color: #f9fafb;
  border: solid 1px #dedeeb;
  border-radius: 4px;
  width: 100%;
  height: 32px;
}

<div
  class="c0"
/>
`;

exports[`styledComponents <Header/> should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .left {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .left.as-column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
/>
`;

exports[`styledComponents <HeaderText/> should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 4px;
}

.c0 .name {
  font-family: 'Roboto',sans-serif;
  font-size: 11px;
  line-height: 16px;
  color: #031A51;
  text-transform: initial;
  font-weight: 400;
}

.c0 .type {
  font-family: 'Roboto',sans-serif;
  font-size: 11px;
  line-height: 16px;
  color: #8697BF;
  text-transform: initial;
  font-weight: 400;
}

<div
  class="c0"
/>
`;

exports[`styledComponents <HeaderText/> should render with size='medium' 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 4px;
}

.c0 .name {
  font-family: 'Roboto',sans-serif;
  font-size: 15px;
  line-height: 24px;
  color: #031A51;
  font-weight: 400;
}

.c0 .type {
  font-family: 'Roboto',sans-serif;
  font-size: 11px;
  line-height: 16px;
  color: #8697BF;
  text-transform: initial;
  font-weight: 400;
}

<div
  class="c0"
/>
`;
